### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by Natim**
@leplatrem r?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+language: python
+python: 2.7
```
Comment (leplatrem) [inline .travis.yml:2]: You can do better ;)

File: tests/test_client.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+import mock
+import pytest
+from portier.client import discover_keys, get_verified_email
```
Comment (leplatrem) [inline tests/test_client.py]: nit: separation between imports

File: tests/test_client.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            {"jwks_uri": "http://broker-url.tld/jwks_uri"},
+            {"keys": []}
+        )
+        keys = discover_keys("http://broker-url.tld/", cache)
+
+        assert isinstance(keys, dict)
+
+        assert mocked_requests.get.call_count == 2
+        mocked_requests.get.assert_any_call(
+            "http://broker-url.tld//.well-known/openid-configuration")
```
Comment (leplatrem) [inline tests/test_client.py]: Double slash ?

File: tests/test_client.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    cache = mock.MagicMock()
+    cache.get.return_value = None
+    with mock.patch("portier.client.requests") as mocked_requests:
+        mocked_requests.get.return_value.json.side_effect = (
+            {"jwks_uri": "http://broker-url.tld/jwks_uri"},
+            {"keys": []}
+        )
+        keys = discover_keys("http://broker-url.tld/", cache)
+
+        assert isinstance(keys, dict)
```
Comment (leplatrem) [inline tests/test_client.py]: I think that part doesn't have to be in the `with`

File: tests/test_client.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    with mock.patch("portier.client.requests") as mocked_requests:
+        mocked_requests.get.return_value.json.return_value = {}
+        with pytest.raises(ValueError) as e:
+            discover_keys("http://broker-url.tld/", cache)
+        assert "No jwks_uri in discovery document" in str(e)
+
+
+def test_discover_key_raises_a_value_error_if_keys_not_found():
+    cache = mock.MagicMock()
+    cache.get.return_value = None
```
Comment (leplatrem) [inline tests/test_client.py]: Declare an `empty_cache` once on top somewhere?

File: tests/test_client.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+
+def test_get_verified_return_the_subject_and_redirect_uri():
+    cache = mock.MagicMock()
+    cache.get.side_effect = (
+        {"keys": [
+            {"kid": "abc",
+             "e": "KHTPnNouCvwROWeIWQkJiw",
+             "n": "ZgKgqvEo_GZMamwy293IvA",
+             "alg": "RS256"}]},
```
Comment (leplatrem) [inline tests/test_client.py]: I have the feeling that tests could be a lot easier to read if those crazy values would be declared once on top (eg. `warm_cache`, `TOKEN`, etc...)

File: tests/test_client.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        "http://redirect_uri"
+    )
+    with mock.patch("portier.client.jwt") as mocked_jwt:
+        mocked_jwt.decode.return_value = {
+            "sub": "foobar@restmail.com",
+            "nonce": "a nonce"
+        }
+        result = get_verified_email("http://broker-url.tld/",
+                                    "eyJraWQiOiAiYWJjIn0.foo.bar",
+                                    "audience", "issuer", cache)
```
Comment (leplatrem) [inline tests/test_client.py]: Add kwargs names?

**Review by leplatrem — CHANGES_REQUESTED** [review]
I found that the tests were a bit heavy to read :)

File: tests/test_client.py (hunk 5 above)
Comment (Natim) [reply tests/test_client.py]: You want me to create a class with a setup phase right?

File: .travis.yml (hunk 1 above)
Comment (Natim) [reply .travis.yml:2]: What do you mean?

Comment (Natim) [reply .travis.yml:2]: It should also work with Py3

**Comment by Natim** [comment]
> I found that the tests were a bit heavy to read :)

Is it better now?

**Review by leplatrem — APPROVED** [review]
Yeah! better :)

**Pull request merged**